pallets/pools/src/lib.rs

```diff
@@ -615,6 +616,23 @@ pub mod pallet {
 			let now = Self::now();
 			let tranches = Tranches::from_input::<T::TrancheToken>(pool_id, tranches, now)?;
 
+			let checked_metadata: Option<BoundedVec<u8, T::MaxSizeMetadata>> = match metadata {
```

Comment:
There must be a better way to do this, open to suggestions!

Reply:
If we consider my point about the event, we can turn this into a `.map` or `.and_then` pipe and clean it up a little.

Reply:
@branan Nuno and I both looked at this but couldn't figure it out, the `try_into` makes this rather difficult. Our conclusion:
> I'd say let it be because the only say I presume we can get it to work is by being explicit about the try_into conversion type and that will end up just bloat the code about as much

Let me know what you think (or if you have a suggestion for a better way of doing this)!

Reply:
Yeah, I think the compiler can't infer what we want here, and the turbofish is probably grosser than the match statement we've got. Let's keep this as-is, and we can always clean it up if a nicer way to write it appears.